Tools/environment_install/install-prereqs-ubuntu.sh

```diff
@@ -97,7 +97,8 @@ if [ -n "$RP" ]; then
 fi
 
 $APT_GET install $BASE_PKGS $SITL_PKGS $PX4_PKGS $ARM_LINUX_PKGS
-pip2 -q install --user -U $PYTHON_PKGS
+
+sudo su - $USER -c "pip2 -q install --user -U $PYTHON_PKGS"
```

Comment:
why do you need that ?

Reply:
See other comment. Docker runs this script as `root`, so pip installs packages for the root user, not `ardupilot`.

Reply:
That modified Dockerfile looks like the way to go to me, thanks.

Why the apt-get install of `gcc-arm-none-eabi`?  The install-prereqs script grabs a specific, vetted, version of the compiler.  Given people may be using the Docker image to build things to flash onto vehicles, sticking with the evil-we-know is probably better.  We absolutely do want to move our compilers forward - but would prefer not to have random developers losing vehicles because we're doing something wrong according to newer compilers!

Reply:
Ok - I'll change the PR.

That gcc install has been in the Dockerfile from the start by the looks of it https://github.com/ArduPilot/ardupilot/commit/2fe3535fd81434dac4146694ed38435e7ed335ec

Reply:
yep !